### ISSUE TITLE
Can not access browser vnc session

### ISSUE THREAD

**Issue by baflQA** [wrike/callisto-chart#6]
Hey - first of all: great job 👍 
I have a problem with accessing VNC session in selenoid UI. 
When I click the unlock button:
![Zrzut ekranu 2022-02-21 o 10 46 37](https://user-images.githubusercontent.com/27357499/154929752-b980b2c5-c81c-4c72-a1eb-5b3f7446e860.png)

the request to:
https://myCallistoDeployment.com/vnc/browser-h6hjc-100.98.130.158-57ebafae17e4bfeaffc92f1e897f8a03

crashes with 502 Bad Gateway. 

My nginx config is as follows:
```upstream callisto {
  server {{ template "callisto.name" . }}:8080;
}
upstream selenoid {
  server {{ template "callisto.name" . }}-selenoid-ui:8000;
}
map $http_upgrade $connection_upgrade {
    default upgrade;
    ''      close;
}

log_format custom '$remote_addr [$time_iso8601] $scheme "$request" $request_time $status "$http_user_agent" $upstream_response_time "$host" $upstream_addr "{{ .Values.instanceID }}"';
access_log off;

server {
  listen 80 default_server;
  resolver {{ .Values.nginx.resolver }};
  access_log /var/log/nginx/access.log custom;

  proxy_connect_timeout       600;
  proxy_send_timeout          600;
  proxy_read_timeout          600;
  send_timeout                600;

  error_page 504 /504.html;
    location /504.html {
    return 504 '{"error": "504", "message": "Gateway Timeout at Callisto nginx. Probably the browser has crashed."}';
  }
  error_page 503 /503.html;
    location /503.html {
    return 503 '{"error": "503", "message": "Service Temporarily Unavailable at Callisto nginx. Probably the Kubernetes API is unavailable."}';
  }
  error_page 502 /502.html;
    location /502.html{
    return 502 '{"error": "502", "message": "Bad Gateway at Callisto nginx. Probably the container for the browser is not running."}';
  }
  location ~ /session/(?<pod_name>browser-[a-z0-9]+)-(?<pod_ip>[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)-(?<session_id>[a-z0-9-]+)/(?<tests_request>.+) {
    proxy_http_version 1.1;
    proxy_connect_timeout       30;
    proxy_send_timeout          30;
    proxy_read_timeout          30;
    send_timeout                30;
    client_max_body_size        8m;
    proxy_request_buffering off; # disable request buffering on disk
    {{- if eq .Values.callisto.browser.name "firefox" }}
    proxy_pass http://$pod_ip:4444{{ .Values.callisto.browser.firefoxWebdriverPath }}/session/$session_id/$tests_request;
    {{- else }}
    proxy_pass http://$pod_ip:4444/session/$session_id/$tests_request;
    {{- end }}
  }
  location ~ /session/(?<pod_name>browser-[a-z0-9]+)-(?<pod_ip>[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)-(?<session_id>[a-z0-9-]+) {
    proxy_http_version 1.1;
    proxy_pass http://callisto/api/v1/session/$pod_name;
  }
  location /session {
    proxy_http_version 1.1;
    proxy_pass http://callisto/api/v1/session;
  }
  location ~ /vnc/(?<pod_name>browser-[a-z0-9]+)-(?<pod_ip>[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)-(?<session_id>[a-z0-9-]+) {
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $connection_upgrade;
    proxy_set_header Host $host;
    proxy_buffering off;
    proxy_http_version 1.1;
    proxy_pass http://$pod_ip:5900;
  }
  location ~ /ws/logs/(?<pod_name>browser-[a-z0-9]+)-(?<pod_ip>[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)-(?<session_id>[a-z0-9-]+) {
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $connection_upgrade;
    proxy_set_header Host $host;
    proxy_buffering off;
    proxy_http_version 1.1;
    proxy_pass http://callisto/api/v1/logs/$pod_name;
  }
  location / {
    # selenoid-ui
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $connection_upgrade;
    proxy_set_header Host $host;
    proxy_buffering off;
    proxy_http_version 1.1;
    proxy_pass http://selenoid/;
  }
  # We don't support video playback in Selenoid-UI
  location ~ /callisto/video {
    return 200 '{}';
  }
  location ~ /callisto/(.*) {
    proxy_http_version 1.1;
    proxy_pass http://callisto/api/v1/$1;
  }
  location /stub_status {
      stub_status;
      access_log off;
      allow all;
  }
  location /health {
    access_log off;

    return 200;
  }
}
```

**Comment by vpokotilov** [comment]
@baflQA hi!

502 may indicate that the VNC server running inside the browser pod is not working properly.

What browser image are you using and what version of Selenoid-UI?

**Comment by baflQA** [comment]
Hi! 
I'm running on selenoid-ui:1.10.4 and selenoid/vnc:chrome_97.0 images.

In the browser's pod console I can see that VNC is running
<img width="1454" alt="vnc" src="https://user-images.githubusercontent.com/27357499/155140979-3d5c6170-130f-4943-97d6-cd022dcf5994.png">

**Comment by vpokotilov** [comment]
Okay, I can reproduce this error.
The latest version of Chrome image that everything works on `selenoid/vnc:chrome_84.0`.
You can use it if it fits.

I'll try to figure out next week why it doesn't work on newer versions.

**Comment by baflQA** [comment]
Hey - perfect, thanks for Your effort. 
I'm still testing callisto, so It's not a blocker for me, will stick to some modern Chrome version then.

**Comment by vpokotilov** [comment]
Results of my research:
The problem occurs in the noVNC client used by Selenoid-UI. There are several issues in the noVNC repository that refer to a bug in the libvncserver library ([issue](https://github.com/LibVNC/libvncserver/issues/297)). Apparently, the bug appeared in version 0.9.12, and it was fixed in version 0.9.13.

I built the image with libvncserver version 0.9.13 using the [instructions](https://aerokube.com/images/latest/#_building_images), and VNC in Callisto works correctly with it.

But all new versions of selenoid browser images use version 0.9.12 (because they are [based on Ubuntu 20.04 LTS](https://github.com/aerokube/images/blob/master/selenium/base/Dockerfile#L18)).

We will try to find some workaround for this problem (maybe we can do it on the nginx side). Theoretically, this is possible, since there are no problems with VNC when using selenoid, on any versions of images.

The recommendation at the moment is to use the `selenoid/vnc:chrome_84.0` image for debugging via VNC.
